micro-service/src/app/[[...slug]]/page.js

```diff
@@ -0,0 +1,66 @@
+import '../../index.css';
+import { ClientOnly } from './client';
+import { headers } from 'next/headers';
+import { getChurchSlug } from '../../utils/index';
+import parseSlugToIdAndType from '@apollosproject/web-shared/utils/parseSlugToIdAndType';
+import Head from 'next/head';
+
+export const dynamic = 'force-dynamic';
+
+export async function generateMetadata({ params, searchParams }, parent) {
+  const id = searchParams?.id;
+
+  if (!id) return parent;
+
+  const route = headers().get('referer');
+
+  const churchSlug = getChurchSlug(route);
+
+  const { type, randomId } = parseSlugToIdAndType(id) ?? [];
+
+  // fetch data
+  const response = await fetch('https://cdn.apollos.app/', {
```

Comment:
A little bummed that we couldn't use apollo query here, but overall I'm not too bothered 🤷